tests/moduletool/test_install.py

```diff
@@ -941,3 +942,163 @@ def test_install_project_with_install_mode_master(tmpdir: py.path.local, snippet
     """.strip() in str(
         excinfo.value
     )
+
+
+def test_module_install_logging(local_module_package_index: str, snippetcompiler_clean, caplog) -> None:
+    """
+    Make sure the module's informations are displayed when it is being installed for both v1 and v2 modules.
+    """
+
+    caplog.set_level(logging.INFO)
+
+    v1_module = "minimalv1module"
+    v2_module = "minimalv2module"
+
+    v2_requirements = [Requirement.parse(module.ModuleV2Source.get_package_name_for(v2_module))]
+
+    # set up project and modules
+    project: module.Project = snippetcompiler_clean.setup_for_snippet(
+        "\n".join(f"import {mod}" for mod in ["std", v1_module, v2_module]),
+        autostd=False,
+        python_package_sources=[local_module_package_index],
+        python_requires=v2_requirements,
+        install_project=False,
+    )
+
+    os.chdir(project.path)
+
+    # autostd=True reports std as an import for any module, thus requiring it to be v2 because v2 can not depend on v1
+    module.Project.get().autostd = False
+    ProjectTool().execute("install", [])
+
+    # ensure we can compile
+    compiler.do_compile()
+
+    expected_logs = [
+        ("Installing module minimalv2module (v2)", logging.INFO),
+        ("Successfully installed module minimalv2module (v2) version", logging.INFO),
+        ("Installing module std (v1)", logging.INFO),
+        ("Successfully installed module std (v1) version", logging.INFO),
+    ]
+
+    for message, level in expected_logs:
+        log_contains(
+            caplog,
+            "inmanta.module",
+            level,
+            message,
+        )
+
+
+def test_real_time_logging(caplog):
+    """
+    Make sure the logging in _run_command_and_stream_output happens in real time
+    """
+    caplog.set_level(logging.DEBUG)
+
+    cmd: List[str] = ["sh -c 'echo one && sleep 1 && echo two'"]
+    return_code: int
+    output: List[str]
+    return_code, output = PythonEnvironment._run_command_and_stream_output(cmd, shell=True)
+    assert return_code == 0
+
+    assert "one" in caplog.records[0].message
+    assert "one" in output[0]
+    first_log_line_time: datetime = datetime.fromtimestamp(caplog.records[0].created)
+
+    assert "two" in caplog.records[-1].message
+    assert "two" in output[-1]
+    last_log_line_time: datetime = datetime.fromtimestamp(caplog.records[-1].created)
+
+    # "two" should be logged at least one second after "one"
+    delta: float = (last_log_line_time - first_log_line_time).total_seconds()
+    assert delta >= 1
+
+
+def test_no_matching_distribution(local_module_package_index: str, snippetcompiler_clean, caplog):
+    """ """
+    caplog.set_level(logging.DEBUG)
+
+    v2_module = "no_matching_dependency"
+
+    v2_requirements = [Requirement.parse(module.ModuleV2Source.get_package_name_for(v2_module))]
+
```

Comment:
Rewrite this using the module_from_template() method to avoid polluting the modules_v2 directory.